src/json-patch-duplex.js

```diff
@@ -513,9 +513,11 @@ var jsonpatch;
         if (obj === undefined) {
             return true;
         }
-        if (typeof obj == "array" || typeof obj == "object") {
-            for (var i in obj) {
-                if (hasUndefined(obj[i])) {
+        if (obj && (_isArray(obj) || typeof obj === "object")) {
+            var objKeys = _objectKeys(obj);
```

Comment:
Getting the object keys of an array will return a (possibly huge) array of incrementing numbers, which are then used as indices to iterate over the array itself. Arrays should be iterated directly